### ISSUE TITLE
Installing Error on ./configure

### ISSUE THREAD

**Issue by zerodwide** [vysheng/tg#806]
i think my python is not right

my os:
 Ubuntu 15.04 64bit Desktop on VMWARE
 memory: 3GB

my log:
 root@ubuntu:/var/www/Telegram/tg# ./configure

checking for gcc... gcc
 checking whether the C compiler works... yes
 checking for C compiler default output file name... a.out
 checking for suffix of executables... 
 checking whether we are cross compiling... no
 checking for suffix of object files... o
 checking whether we are using the GNU C compiler... yes
 checking whether gcc accepts -g... yes
 checking for gcc option to accept ISO C89... none needed
 checking for sqrt in -lm... yes
 checking for library containing clock_gettime... none required
 checking for library containing backtrace... none required
 checking for event_base_new in -levent... yes
 checking how to run the C preprocessor... gcc -E
 checking for grep that handles long lines and -e... /bin/grep
 checking for egrep... /bin/grep -E
 checking for ANSI C header files... yes
 checking for sys/types.h... yes
 checking for sys/stat.h... yes
 checking for stdlib.h... yes
 checking for string.h... yes
 checking for memory.h... yes
 checking for strings.h... yes
 checking for inttypes.h... yes
 checking for stdint.h... yes
 checking for unistd.h... yes
 checking event2/event.h usability... yes
 checking event2/event.h presence... yes
 checking for event2/event.h... yes
 checking for pkg-config... /usr/bin/pkg-config
 checking whether compiling and linking against OpenSSL works... yes
 checking if zlib is wanted... yes
 checking for inflateEnd in -lz... yes
 checking zlib.h usability... yes
 checking zlib.h presence... yes
 checking for zlib.h... yes
 checking for rl_save_prompt in -lreadline... yes
 checking for libconfig... enabled
 checking for config_init in -lconfig... yes
 checking for liblua... enabled
 checking for lua... /usr/bin/lua
 checking if /usr/bin/lua is a Lua interpreter... yes
 checking for lua version... 5.2
 checking for lua platform... unknown
 checking for lua script directory... ${prefix}/share/lua/5.2
 checking for lua module directory... ${exec_prefix}/lib/lua/5.2
 checking if LUA_VERSION is defined... yes
 checking lua.h usability... no
 checking lua.h presence... no
 checking for lua.h... no
 checking lualib.h usability... no
 checking lualib.h presence... no
 checking for lualib.h... no
 checking lauxlib.h usability... no
 checking lauxlib.h presence... no
 checking for lauxlib.h... no
 checking luaconf.h usability... no
 checking luaconf.h presence... no
 checking for luaconf.h... no
 checking for Lua headers in... /usr/include/lua5.2
 checking lua.h usability... yes
 checking lua.h presence... yes
 checking for lua.h... yes
 checking lualib.h usability... yes
 checking lualib.h presence... yes
 checking for lualib.h... yes
 checking lauxlib.h usability... yes
 checking lauxlib.h presence... yes
 checking for lauxlib.h... yes
 checking luaconf.h usability... yes
 checking luaconf.h presence... yes
 checking for luaconf.h... yes
 checking for Lua header version... 5.2
 checking if Lua header version matches 5.2... yes
 checking if LUA_VERSION is defined... yes
 checking for library containing exp... none required
 checking for library containing dlopen... -ldl
 checking for library containing lua_load... -llua5.2
 checking for python... enabled
 checking for python build information... 
 checking for python3.4... python3.4
 checking for main in -lpython3.4... no
 checking for main in -lpython3.4m... no
 checking for main in -lpython3.4mu... no
 checking for python3.3... no
 checking for python3.2... no
 checking for python3.1... no
 checking for python3... python3
 checking for main in -lpython3... no
 checking for main in -lpython3m... no
 checking for main in -lpython3mu... no
 checking for python2.7... python2.7
 checking for main in -lpython2.7... yes
 results of the Python check:
 Binary: python2.7
 Library: python2.7
 Include Dir: /usr/include/python2.7
 checking for libjansson... enabled
 checking for json_array_set_new in -ljansson... yes
 checking progname... default
 checking execinfo.h usability... yes
 checking execinfo.h presence... yes
 checking for execinfo.h... yes
 checking fcntl.h usability... yes
 checking fcntl.h presence... yes
 checking for fcntl.h... yes
 checking malloc.h usability... yes
 checking malloc.h presence... yes
 checking for malloc.h... yes
 checking netdb.h usability... yes
 checking netdb.h presence... yes
 checking for netdb.h... yes
 checking for stdlib.h... (cached) yes
 checking for string.h... (cached) yes
 checking for unistd.h... (cached) yes
 checking arpa/inet.h usability... yes
 checking arpa/inet.h presence... yes
 checking for arpa/inet.h... yes
 checking mach/mach.h usability... no
 checking mach/mach.h presence... no
 checking for mach/mach.h... no
 checking netinet/in.h usability... yes
 checking netinet/in.h presence... yes
 checking for netinet/in.h... yes
 checking sys/file.h usability... yes
 checking sys/file.h presence... yes
 checking for sys/file.h... yes
 checking sys/socket.h usability... yes
 checking sys/socket.h presence... yes
 checking for sys/socket.h... yes
 checking termios.h usability... yes
 checking termios.h presence... yes
 checking for termios.h... yes
 checking for size_t... yes
 checking for uid_t in sys/types.h... yes
 checking for inline... inline
 checking for stdlib.h... (cached) yes
 checking for GNU libc compatible malloc... yes
 checking for stdlib.h... (cached) yes
 checking for GNU libc compatible realloc... yes
 checking for alarm... yes
 checking for endpwent... yes
 checking for memset... yes
 checking for memmove... yes
 checking for mkdir... yes
 checking for select... yes
 checking for socket... yes
 checking for strdup... yes
 checking for strndup... yes
 checking for uname... yes
 checking for __builtin_bswap32... yes
 configure: creating ./config.status
 config.status: creating Makefile
 config.status: creating config.h
 config.status: config.h is unchanged

root@ubuntu:/var/www/Telegram/tg# make
 make: *** No rule to make target 'tgl/scheme.tl', needed by 'auto/scheme.tl'. Stop.

**Comment by zerodwide** [comment]
i looked at same issues, their problem related to swap 
 -> vysheng/tg#463
 and i did that instruction, but still have problem!

i have something more than those logs

mine:
 checking for python... enabled
 checking for python build information...
 checking for python3.4... python3.4
 checking for main in -lpython3.4... no
 checking for main in -lpython3.4m... no
 checking for main in -lpython3.4mu... no
 checking for python3.3... no
 checking for python3.2... no
 checking for python3.1... no
 checking for python3... python3
 checking for main in -lpython3... no
 checking for main in -lpython3m... no
 checking for main in -lpython3mu... no
 checking for python2.7... python2.7
 checking for main in -lpython2.7... yes
 results of the Python check:
 Binary: python2.7
 Library: python2.7
 Include Dir: /usr/include/python2.7

their:
 default

**Comment by zerodwide** [comment]
checking for my python version:

root@ubuntu:/home/tx# python -V
 Python 2.7.9
 root@ubuntu:/home/tx# readlink -f $(which python)
 /usr/bin/python2.7

**Comment by zerodwide** [comment]
sorry i had downloaded repo by zip 
and tgl/* was missed!

i ran this, and no error occurred 
git clone --recursive https://github.com/vysheng/tg.git && cd tg

**Comment by psycholyzern** [comment]
yep, you need ```--recursive``` to download sub module too. Please close issue if solved

**Comment by zerodwide** [comment]
perfectly solved. please close issue

**Comment by psycholyzern** [comment]
close it yourself :)

**Comment by taa1** [comment]
Is there a way to download tg-master.zip and not have it give the error:

`make: *** No rule to make target 'tgl/scheme.tl', needed by 'auto/scheme.tl'. Stop.`

To put it another way, is git the _only_ way to download tg?

**Comment by psycholyzern** [comment]
@taa1, I guess you can download tgl separately and put in the tg folder.
